CRM/Contribute/Form/AbstractEditPayment.php

```diff
@@ -562,10 +562,10 @@ protected function beginPostProcess() {
       if (in_array('credit_card_exp_date', array_keys($this->_params))) {
         $this->_params['year'] = CRM_Core_Payment_Form::getCreditCardExpirationYear($this->_params);
         $this->_params['month'] = CRM_Core_Payment_Form::getCreditCardExpirationMonth($this->_params);
+        $this->assign('credit_card_exp_date', CRM_Utils_Date::mysqlToIso(CRM_Utils_Date::format($this->_params['credit_card_exp_date'])));
```

Comment:
So in general I prefer to ensure 'something' is assigned to the template - even if blank - this generally reduces leakage. In this case it's a form function so it's not clear leakage could occur. So I'm a soft preference for changing that.

I'd also prefer not to have params other than credit_card_exp_date in the if for that field. I know you can argue that there wouldn't be a card number with no date & I can't think of a case where that would be false but we are still using one field as a stand in for a greater meaning - I think

```
  $this->assign('credit_card_type', CRM_Utils_Array::value('credit_card_type', $this->_params));
```

per below is better practice

Reply:
Yes. I messed this up - have pushed an updated version of `assignCreditCardDetails` which always assigns.